improver/regrid/grid.py

```diff
@@ -66,8 +66,8 @@ def calculate_input_grid_spacing(cube_in: Cube) -> Tuple[float, float]:
         raise ValueError("Input grid is not on a latitude/longitude system")
 
     # calculate grid spacing
-    lon_spacing = calculate_grid_spacing(cube_in, "degree", axis="x", rtol=1.0e-5)
-    lat_spacing = calculate_grid_spacing(cube_in, "degree", axis="y", rtol=1.0e-5)
+    lon_spacing = calculate_grid_spacing(cube_in, "degree", axis="x", rtol=7.0e-5)
+    lat_spacing = calculate_grid_spacing(cube_in, "degree", axis="y", rtol=7.0e-5)
```

Comment:
@gavinevans Hi Gavin, Thank you for your reviewing.

The problem happened when I ran neighbourhood process and regridding. There is an inherent problem in the grid-spacing rtol, which is related to dtype and grid resolution .  Any functions related to calculate_grid_spacing will have this potential problem. My fixing is temporary for several BOM-used grids, and a better fixing is probably required. 

As IMPROVER is using float32 ,  it means only 5 or less significant digits for grid_spacing for current regular grid resolution.    
For example,  longitutde coord: 152.12345, ..... degree, grid size: about 0.2 degree level. Grid spacing = np.diff(coord): 0.20000 or 0.19999 due to rounding error.  rtol=1/20000 =5.0e-5.  With smaller resolution, the function could fail again.

Reply:
Thanks for the example, @zfan001 

I think that the potential need to modify the relative tolerance in grid.py is fair due to the different grids that could be processed. I think that either there is a need to pass in an acceptable `grid_spacing_relative_tolerance` to the CLI, as I've suggested, or alternatively, compute an acceptable value for the relative tolerance based on the data provided somehow, although I'm not exactly sure what would be sensible with an automated approach.

Reply:
Given that the new unit test shows you need an `rtol` of 3e-5, can we use that value here? Or are the real grids just slightly messier, so that a more tolerant value is required?
Your point about finer resolutions needing higher tolerances is valid, but only up to a point. I don't believe anybody would define a grid like that. Regular grids have anchors at whole numbers, e.g. 152.0000. The only rational number likely to be used for an interval that cannot be represented without recurrence is 1/3 (or 1/6, 1/9) as I cannot see anyone going for 1/7. 
The next logical step, if resolution increases further, is to increase the precision of the coordinate to float64, at which point the rtol value could be lower again.

Reply:
Agreed,  I changed 7.0e-5 to to 3.0e-5.